### DISCUSSION TITLE
tracking-down-a-hanging-xunit-test-in-ci-building-a-custom-test-framework/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#370] [Announcements]
# tracking-down-a-hanging-xunit-test-in-ci-building-a-custom-test-framework/

In this post I describe how we tracked down a hanging xUnit test by building a custom XuniTestFramework implementation

https://andrewlock.net/tracking-down-a-hanging-xunit-test-in-ci-building-a-custom-test-framework/

[a comment by an automated account is not shown]

**Comment by siva538** [comment]
Spot ON! This is exactly I was looking for!! Many thanks @Andrew.

**Comment by mhooshmand** [comment]
Hi Andrew,
Thanks for your useful post.

I was thinking of something like GlobalErrorHandling in asp.net for catching all errors raised by Assert() in a unit of code.
in order to :
1. have one code block to log info about test failed
2. access to parameters of test method (in any shape: InlineData, ClassData, MemberData), if it's possible?

I started by talking with GPT, found some thing to search on google and finally found your useful post.

I encounter a probelm.
In put a breakpoint in the below mehtod:
   protected override async Task<RunSummary> RunTestCaseAsync(IXunitTestCase testCase)

For every test method this break point hit once for mehtod with ClassData.
Where is the mehtod that run for every member of the class?

Thanks for your attention

**Comment by kerryjiang** [comment]
Xunit.v3 is different.

It seems not easy to do same thing with Xunit.v3.

**Comment by evaapavel** [comment]
Hi Andrew,

thank you very much. This is exactly what I needed and could not find elsewhere.

ChatGPT yielded "Too many concurrent requests" and went silent...

Fortunately, I happened to spot your post among other Google results and I was saved.

The comment by @kerryjiang makes me a little bit nervous but I hope we will migrate this into xUnit 3 somehow, too.
